sandy/sections/mf2.py

```diff
@@ -182,272 +186,333 @@ def read_mf2(tape, mat):
                         LRU1_LRF1_2_NRO0 = {}
                         LRU1_LRF1_2_NRO0.update(header2)
                         C, i = sandy.read_cont(df, i)
                         add = {
-                                "SPI": C.C1,
-                                "AP": C.C2,
-                                }
+                            "SPI": C.C1,
+                            "AP": C.C2,
+                        }
                         NLS = int(C.N1)
                         LRU1_LRF1_2_NRO0.update(add)
                         LRU1_LRF1_2_NRO0_NLS = {}
-                        for k in range(NLS) :
+                        for k in range(NLS):
                             L, i = sandy.read_list(df, i)
                             add = {
-                                    "AWRI": L.C1,       #Ratio of the mass of a particular isotope to that of a neutron
-                                    "QX": L.C2,         #Q-value to be added to the incident particle’s center-of-mass energy to determine the channel energy for use in the penetrability factor.
-                                    "L": L.L1,          #Value of l.
-                                    "LRX": L.L2,        #Flag indicating whether this energy range contains a competitive width
-                                    "NRS": L.N2,        #Number of resolved resonances for a given l-value.
-                                    }
+                                "AWRI": L.C1,  # Ratio of the mass of a particular isotope to that of a neutron
+                                # Q-value to be added to the incident
+                                # particle’s center-of-mass energy to determine
+                                # the channel energy for use in the
+                                # penetrability factor.
+                                "QX": L.C2,
+                                "L": L.L1,  # Value of l.
+                                "LRX": L.L2,  # Flag indicating whether this energy range contains a competitive width
+                                # Number of resolved resonances for a given
+                                # l-value.
+                                "NRS": L.N2,
+                            }
                             keys = ["ER", "AJ", "GT", "GN", "GG", "GF"]
-                            RES_PAR = [dict(zip(keys, items)) for items in sandy.utils.grouper(L.B, 6)]
-                            add.update({"RES_PAR":RES_PAR})
-                            LRU1_LRF1_2_NRO0_NLS.update({k:add})
-                            LRU1_LRF1_2_NRO0.update({"L":LRU1_LRF1_2_NRO0_NLS})
-                        dico[(EL, EH)] = LRU1_LRF1_2_NRO0  
-                    else :   
+                            RES_PAR = [dict(zip(keys, items))
+                                       for items in sandy.utils.grouper(L.B, 6)]
+                            add.update({"RES_PAR": RES_PAR})
+                            LRU1_LRF1_2_NRO0_NLS.update({k: add})
```

Comment:
The key should not be k, but L.L1. We never use an iterating pointer as key of a dictionary!